### PR TITLE
introduce state of a resource to Resource in types.

### PR DESCRIPTION
**Pull request by vaikas**
This is in preparation for #79 and #80. Want us to agree on the representation before moving further.

### CODE REVIEW

**Comment by vaikas-google** [comment]
@jackgr 
@bmelville

File: manager/manager/types.go [hunk 1]
```diff
@@ -140,6 +140,23 @@ type Configuration struct {
...
+// These constants implement the resourceStatus enumeration type.
+const (
+	Created ResourceStatus = "Created"
+	Failed  ResourceStatus = "Failed"
+	Aborted ResourceStatus = "Aborted"
+)
+
+// ResourceState describes the state of a resource.
+type ResourceState struct {
+	Status   ResourceStatus `json:"status"`
```
Comment (bmelville) [inline manager/manager/types.go]: What is the default here? Either have a pending/inprogress default or mark it omitempty as well

Comment (vaikas-google) [inline manager/manager/types.go]: Yes, there's always one of the states there. Added omitempty and a comment that it's a terminal state. PTAL.

File: manager/manager/types.go [hunk 2]
```diff
@@ -84,7 +84,7 @@ func NewDeployment(name string, id int) *Deployment {
 // DeploymentStatus is an enumeration type for the status of a deployment.
 type DeploymentStatus string
 
-// These constants implement the deploymentStatus enumeration type.
+// These constants implement the DeploymentStatus enumeration type.
 const (
 	CreatedStatus  DeploymentStatus = "Created"
 	DeletedStatus  DeploymentStatus = "Deleted"
```
Comment (jackgr) [inline manager/manager/types.go]: Shouldn't the two preceding status strings be deleted, now that we have the ones below?

Comment (jackgr) [inline manager/manager/types.go]: Never mind. I see that these are deployment status, while the others are resource status. Makes sense.

**Pull request merged**